### PR TITLE
[INLONG-3033][Manager] Add sourceFieldName in sinkFields

### LINKED ISSUE #3033: [Improve] Support source Field in Manager Client
### Description

_No response_

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kipshi**
### Title Name: [INLONG-3033] Add sourceFieldName in sinkFields

where *XYZ* should be replaced by the actual issue number.

Fixes #3033 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 1]
```diff
@@ -244,21 +230,32 @@ private static HiveSinkRequest createHiveRequest(StreamSink streamSink, InlongSt
...
-                    })
-                    .collect(Collectors.toList());
+            List<SinkFieldRequest> fieldRequests = createSinkFields(streamSink.getStreamFields(),
+                    streamInfo.getFieldList());
             hiveSinkRequest.setFieldList(fieldRequests);
         }
         return hiveSinkRequest;
     }
 
+    private static List<SinkFieldRequest> createSinkFields(List<StreamField> sinkFields,
```
Comment (healchow) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:240]: The `StreamField` may be named `SinkField`?

Comment (kipshi) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:240]: not necessary, StreamField is class for field definition in Client

File: inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java [hunk 2]
```diff
@@ -244,21 +230,32 @@ private static HiveSinkRequest createHiveRequest(StreamSink streamSink, InlongSt
...
+            if (sourceFields.size() > i && sourceFields.get(i) != null) {
+                InlongStreamFieldInfo sourceField = sourceFields.get(i);
+                sinkFieldRequest.setSourceFieldName(sourceField.getFieldName());
+                sinkFieldRequest.setSourceFieldType(sourceField.getFieldType());
+            }
+            sinkFieldRequests.add(sinkFieldRequest);
+        }
+        return sinkFieldRequests;
+    }
+
```
Comment (gong) [inline inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:258]: Is field mapping created by order of sink field list and stream field list ?

Comment (kipshi) [reply inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/util/InlongStreamSinkTransfer.java:258]: It is now, and yunqingmo will add Mappings after this pr is merged

**Review by healchow — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Pull request merged**